pkg/controller/ingress-shim/controller.go

```diff
@@ -72,41 +67,35 @@ type controller struct {
 // It returns the workqueue to be used to enqueue items, a list of
 // InformerSynced functions that must be synced, or an error.
 func (c *controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitingInterface, []cache.InformerSynced, error) {
-	// construct a new named logger to be reused throughout the controller
-	c.log = logf.FromContext(ctx.RootContext, ControllerName)
+	kShared := ctx.KubeSharedInformerFactory
+	cmShared := ctx.SharedInformerFactory
 
-	// create a queue used to queue up items to be processed
-	c.queue = workqueue.NewNamedRateLimitingQueue(controllerpkg.DefaultItemBasedRateLimiter(), ControllerName)
+	c.log = logf.FromContext(ctx.RootContext, ControllerName)
+	queue := workqueue.NewNamedRateLimitingQueue(controllerpkg.DefaultItemBasedRateLimiter(), ControllerName)
 
-	// obtain references to all the informers used by this controller
-	ingressInformer := ctx.KubeSharedInformerFactory.Networking().V1beta1().Ingresses()
-	certificatesInformer := ctx.SharedInformerFactory.Certmanager().V1().Certificates()
-	issuerInformer := ctx.SharedInformerFactory.Certmanager().V1().Issuers()
-	// build a list of InformerSynced functions that will be returned by the Register method.
-	// the controller will only begin processing items once all of these informers have synced.
 	mustSync := []cache.InformerSynced{
-		ingressInformer.Informer().HasSynced,
-		certificatesInformer.Informer().HasSynced,
-		issuerInformer.Informer().HasSynced,
+		cmShared.Certmanager().V1().Certificates().Informer().HasSynced,
+		cmShared.Certmanager().V1().Certificates().Informer().HasSynced,
+		cmShared.Certmanager().V1().Issuers().Informer().HasSynced,
 	}
 
-	// set all the references to the listers for used by the Sync function
-	c.ingressLister = ingressInformer.Lister()
-	c.certificateLister = certificatesInformer.Lister()
-	c.issuerLister = issuerInformer.Lister()
+	c.ingressLister = kShared.Networking().V1beta1().Ingresses().Lister()
+	c.certificateLister = cmShared.Certmanager().V1().Certificates().Lister()
+	c.issuerLister = cmShared.Certmanager().V1().Issuers().Lister()
```

Comment:
This has not changed, we still use the shared cache. Both approaches were returning the same thing:

```go
# Before:
infomer := cmShared.Certmanager().V1().Issuers()
lister := informer.Lister()

# After:
lister := cmShared.Certmanager().V1().Issuers().Lister()
```

(note that I removed the Issuer lister variables since they were unused)